src/app/services/custom-sort/table-custom-sort-functions.service.ts

```diff
@@ -0,0 +1,22 @@
+import { Injectable } from '@angular/core';
+import { Order } from '@darts-types/data-table-column.interface';
+import { TranscriptionUrgency } from '@darts-types/transcription-urgency.interface';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class TableCustomSortFunctionsService {
+  // only works for transcription data table
+  sortByUrgencyPriorityOrder(a: unknown, b: unknown, direction?: Order) {
```

Comment:
Can you not specify the types in the parameters?

```suggestion
  sortByUrgencyPriorityOrder(a: { urgency: TranscriptionUrgency }, b: { urgency: TranscriptionUrgency }, direction?: Order) {
```

Reply:
if we specify in the parameters it breaks the constraints of the interface we have in place so we have to do it this way sadly and type cast afterwards inside the function